src/test/regress/expected/single_shard_table_udfs.out

```diff
@@ -0,0 +1,207 @@
+     1
+(1 row)
+
+COMMIT;
+SELECT partmethod, repmodel FROM pg_dist_partition WHERE logicalrelid = 'null_dist_key_table'::regclass;
+ partmethod | repmodel
+---------------------------------------------------------------------
+ n          | s
+(1 row)
+
+SELECT master_get_table_ddl_events('null_dist_key_table');
+                               master_get_table_ddl_events
+---------------------------------------------------------------------
+ CREATE TABLE null_dist_key_udfs.null_dist_key_table (a integer) USING heap
+ ALTER TABLE null_dist_key_udfs.null_dist_key_table OWNER TO postgres
+ CREATE INDEX null_dist_key_idx ON null_dist_key_udfs.null_dist_key_table USING btree (a)
+(3 rows)
+
+SELECT column_to_column_name(logicalrelid, partkey)
+FROM pg_dist_partition WHERE logicalrelid = 'null_dist_key_table'::regclass;
+ column_to_column_name
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT column_name_to_column('null_dist_key_table', 'a');
+                                                  column_name_to_column
+---------------------------------------------------------------------
+ {VAR :varno 1 :varattno 1 :vartype 23 :vartypmod -1 :varcollid 0 :varlevelsup 0 :varnosyn 1 :varattnosyn 1 :location -1}
+(1 row)
+
+SELECT master_update_shard_statistics(shardid)
+FROM (SELECT shardid FROM pg_dist_shard WHERE logicalrelid='null_dist_key_table'::regclass) as shardid;
+ master_update_shard_statistics
+---------------------------------------------------------------------
+                           8192
+(1 row)
+
+-- will always be no-op as we create the shell table from scratch
+-- while creating a citus local table, but let's see it works
+SELECT truncate_local_data_after_distributing_table('null_dist_key_table');
+ truncate_local_data_after_distributing_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- should return a single element array that only includes its own shard id
+SELECT shardid=unnest(get_colocated_shard_array(shardid))
+FROM (SELECT shardid FROM pg_dist_shard WHERE logicalrelid='null_dist_key_table'::regclass) as shardid;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+BEGIN;
+  SELECT master_remove_partition_metadata('null_dist_key_table'::regclass::oid, 'null_dist_key_udfs', 'null_dist_key_table');
+ master_remove_partition_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+  -- should print 0
+  select count(*) from pg_dist_partition where logicalrelid='null_dist_key_table'::regclass;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+ROLLBACK;
+-- should fail --
+SELECT update_distributed_table_colocation('null_dist_key_table', colocate_with => 'none');
+ERROR:  relation null_dist_key_table should be a hash distributed table
+SELECT master_create_empty_shard('null_dist_key_table');
```

Comment:
Let's have one-liner tests to show that below three fail when the input is a single-shard table:

* create_distributed_table
* create_reference_table
* citus_add_local_table_to_metadata

Reply:
I think we already have these. Could you please take a look and ping me if you mean anything other than the ones I've added